app/helpers/repository_helper.rb

```diff
@@ -169,19 +169,19 @@ def remove_folders(folders, user, repo, path: '/', txn: nil)
         if obj.is_a? Repository::RevisionFile
           files << path
         else
-          dirs[path] = current_revision
+          dirs << path
         end
       end
-      dirs[folder_path] = current_revision
+      dirs << folder_path
     end
-    success, file_messages = remove_files(files, user, repo, path: '', txn: txn)
+    success, file_messages = remove_files(files.reverse, user, repo, path: '', txn: txn, option: 'Folder')
 
     return [success, file_messages] unless success
 
     # folders are removed in git if their contents are removed
-    unless repo.is_a? GitRepository
-      dirs.each do |dir, revision|
-        txn.remove(dir, revision)
+    if repo.is_a? GitRepository
```

Comment:
This will remove directories _only_ if we're using git repositories won't it? What will happen when we use subversion or memory repositories?